File: codechef/chef_and_easy_queries/ChefAndEasyQueries.scala
import scala.io.StdIn.readLine

object ChefAndEasyQueries extends App {
  def run(): Unit = {
    val firstLine = readLine().split(" ")
    val daysUntilNone = firstLine(0).toInt
    val maxQueriesPerDay = firstLine(1).toInt
    println("Days until no queries: " + daysUntilNone)
    println("Number of queries that can be handled per day: " + maxQueriesPerDay)

    val numberOfQueriesPerDay = readLine().split(" ")
    println(numberOfQueriesPerDay.mkString(" "))

    val day = 0

    for(day <- 1 to daysUntilNone) {

    }

    println(day)
  }

  for (testCase <- 1 to readLine().toInt) {
    run()
  }
}

File: codechef/chef_and_easy_queries/ChefAndEasyQueries.php
<?php

$numberOfTestCases = (int)getLine();

for($i = 0; $i < $numberOfTestCases; $i++) {
    run();
}

function getLine() {
    return fgets(STDIN);
}

function run() {
    $caseInfo = explode(' ', getline());
    $daysUntilNone = (int)$caseInfo[0];
    $maxQueriesPerDay = (int)$caseInfo[1];
    $queriesPerDay = explode(' ', getline());

    $queriesTodo = $day = 0;
    for(; $day < $daysUntilNone; $day++) {
        $queriesTodo += (int)$queriesPerDay[$day];
        if ($queriesTodo < $maxQueriesPerDay) {
            echo ++$day . PHP_EOL;
            return;
        }
        $queriesTodo -= $maxQueriesPerDay;
    }

    if ($queriesTodo > 0) {
        $day += floor($queriesTodo/ $maxQueriesPerDay);
    }

    echo ++$day . PHP_EOL;
    return;
}